apps/scan/frontend/src/components/reprint_report_button.tsx

```diff
@@ -0,0 +1,66 @@
+import { Button } from '@votingworks/ui';
+import { assert } from '@votingworks/basics';
+import type { PollsTransition } from '@votingworks/scan-backend';
+import { getConfig, getMachineConfig, getScannerResultsByParty } from '../api';
+import { printReport } from '../utils/print_report';
+
+interface ReprintReportButtonProps {
+  printerInfo?: KioskBrowser.PrinterInfo;
+  lastPollsTransition?: PollsTransition;
+  scannedBallotCount: number;
+  reportLabel: 'Additional' | 'Previous';
+  beforePrint: () => void;
+  afterPrint: (numPages: number) => void;
```

Comment:
I'm not sure I see the benefit of abstracting the data loading and logic for reprinting into this special button component, since the poll worker screen already has the data needed for printing polls transition reports in general.

It's a little confusing to track the control flow with the `beforePrint` and `afterPrint` callbacks. I think the previous approach of having a function defined in the poll worker screen component to do the printing logic was a bit more straightforward.

Reply:
I like the idea of breaking this screen into smaller components. I wonder if there's a way to do it with screens rather than buttons though. Anyway, not a blocking issue

Reply:
Yeah, each part of the flow could be a different screen. That would probably be best. And it will become a heck of a lot easier once certain pieces (*cough* printing) are on the backend and we can just hit endpoints rather than have elaborate functions.